lightningd/options.c

```diff
@@ -309,6 +309,10 @@ static void config_register_opts(struct lightningd *ld)
 	opt_register_arg("--fee-base", opt_set_u32, opt_show_u32,
 			 &ld->config.fee_base,
 			 "Millisatoshi minimum to charge for HTLC");
+	opt_register_arg("--rescan", opt_set_s32, opt_show_s32,
+			 &ld->config.rescan,
+			 "Number of blocks to rescan from the current head, or "
+			 "absolute blockheight if negative");
```

Comment:
Seems more sensible, to use negative to mean "current blockheight minus absolute value" and to use positive for "absolute blockheight".  And to use 0 to mean "current blockheight, no rescan".  Similar to Python arrays/strings, with the "end" being the current known blockheight and a negative index being relative from the end.

But this comment is not a blocking one and I think the current standard is reasonable.

Reply:
I preferred making the more complex case, i.e., choosing a fixed offset, the case with the added sign. Everybody should just use relative offsets from their current head, in particular if they run from a startup script, and only use the absolute offset (with negation) in a recovery scenario.

Reply:
Yeah, nobody should be using this anyway, so I think -ve for relative is pretty common.  Otherwise you want --rescan-absolute or something...